mesonbuild/environment.py

```diff
@@ -1386,3 +1409,26 @@ def parse_entry(cls, entry):
             ccache = []
         # Return value has to be a list of compiler 'choices'
         return compiler, ccache
+
+    def lookup_entry(self, name):
+        """Lookup binary
+
+        Returns command with args as list if found, Returns `None` if nothing is
+        found.
+
+        First tries looking in explicit map, then tries environment variable.
+        """
+        # Try explict map, don't fall back on env var
+        command = self.binaries.get(name)
+        if command is not None:
+            command = mesonlib.stringlistify(command)
+            # Relies on there being no "" env var
+            evar = self.__class__.evarMap.get(name, "")
```

Comment:
there's no need to use `self.__class__` here, just call `self.evarMap`, pythons MRO lookup will do the right thing.

Reply:
And the other cases in this function as well.